cmd/check-reaper/main.go

```diff
@@ -195,3 +220,58 @@ func deletePod(ctx context.Context, client *kubernetes.Clientset, pod v1.Pod) er
 	options := metav1.DeleteOptions{PropagationPolicy: &propagationForeground}
 	return client.CoreV1().Pods(pod.Namespace).Delete(ctx, pod.Name, options)
 }
+
+// podConditions returns true if conditions are met to be deleted for checker pod
+func podConditions(pod v1.Pod, duration float64, phase v1.PodPhase) bool {
+	if time.Now().Sub(pod.CreationTimestamp.Time).Hours() > duration && pod.Status.Phase == phase {
+		log.Infoln("Found pod older than ", duration, " hours in status ", phase, " .Deleting pod:", pod)
+		return true
+	}
+	return false
+}
+
+// jobConditions returns true if conditions are met to be deleted for khjob
+func jobConditions(job khjobcrd.KuberhealthyJob, duration time.Duration, phase khjobcrd.JobPhase) bool {
+	if time.Now().Sub(job.CreationTimestamp.Time) > duration && job.Spec.Phase == phase {
+		log.Infoln("Found khjob older than", duration, "minutes in status", phase)
+		return true
+	}
+	return false
+}
+
+// KHJobDelete fetches a list of khjobs in a namespace and will delete them if they meet given criteria
+func khJobDelete(client *khjobcrd.KHJobV1Client) error {
+
+	opts := metav1.ListOptions{}
+	del := metav1.DeleteOptions{}
+
+	// convert JobDeleteMinutes into time.Duration
+	JobDeleteTimeDuration, err := time.ParseDuration(JobDeleteTimeDurationEnv)
```

Comment:
nitpick!
```suggestion
	jobDeleteTimeDuration, err := time.ParseDuration(JobDeleteTimeDurationEnv)
```

Reply:
good catch I will make this not a global